src/tools/clippy/tests/ui/await_holding_lock.stderr

```diff
@@ -55,80 +55,97 @@ LL |
 LL |         let third = baz().await;
    |                           ^^^^^
 
+error: this `MutexGuard` is passed in through an argument or captured by the closure body
+  --> tests/ui/await_holding_lock.rs:64:54
+   |
+LL |       pub async fn equally_bad(x: MutexGuard<'_, u32>) {
+   |  ______________________________________________________^
+LL | |
+LL | |         drop(x);
+LL | |     }
+   | |_____^
+   |
+   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
+note: these are all the `await` points this lock is held through
+  --> tests/ui/await_holding_lock.rs:64:54
+   |
+LL |     pub async fn equally_bad(x: MutexGuard<'_, u32>) {
```

Comment:
The wording `these are all the `await` points` is kind of shared across all the suspension point. The very first suspension point is, yes, the construction of the async state machine/coroutine. You may think of it as the implicit first `.await` point.

Reply:
Forgive me for tagging for reviewing again without a new change-set.

The lint triggers in this case because a lock guard is assigned into the coroutine/async body. I personally think the lint message here makes sense because the idea is to deny the passing of non-`async` `MutexGuard`s into `async` bodies.

I think the wording needs some adjustment.

Reply:
Wording adjusted. Another clippy test is added to show that it applies to `std::cell::Ref` as well.

Reply:
I am figuring out how to extract span for the upvars. One moment.

Reply:
Now the note in this case is a bit pointless. The error message says "this argument is passed" and points to the argument and the note afterwards says the exact same thing again with a different span. Could you remove the note in this case?